src/goto-diff/unified_diff.cpp

```diff
@@ -148,7 +148,7 @@ void unified_difft::lcss(
   const irep_idt &identifier,
   const goto_programt &old_goto_program,
   const goto_programt &new_goto_program,
-  differencest &differences) const
+  differencest &differences)
```

Comment:
Rather than modifying `differences` in place, would it be possible to have these two functions return a new `differencest`?

Reply:
Sure. I will make it a separate commit and change title of the PR since this PR was only about addition of static. Would it be OK with you if I throw in a small commit about ```expr2javat``` and ask Chris to review?